datalake/src/main/java/com/sequenceiq/datalake/flow/detach/SdxDetachActions.java

```diff
@@ -1,173 +1,297 @@
-    public Action<?, ?> finishedAction() {
-        return new AbstractSdxAction<>(SdxDetachSuccessEvent.class) {
+    @Bean(name = "SDX_ATTACH_NEW_CLUSTER_FAILED_STATE")
+    public Action<?, ?> sdxAttachNewClusterFailedAction() {
+        return new AbstractSdxAction<>(SdxDetachFailedEvent.class) {
             @Override
-            protected SdxContext createFlowContext(FlowParameters flowParameters, StateContext<FlowState, FlowEvent> stateContext,
-                    SdxDetachSuccessEvent payload) {
+            protected SdxContext createFlowContext(FlowParameters flowParameters,
+                    StateContext<FlowState, FlowEvent> stateContext,
+                    SdxDetachFailedEvent payload) {
                 return SdxContext.from(flowParameters, payload);
             }
 
             @Override
-            protected void doExecute(SdxContext context, SdxDetachSuccessEvent payload, Map<Object, Object> variables) throws Exception {
-                LOGGER.info("SDX detach finalized: {}", payload.getResourceId());
-                sdxStatusService.setStatusForDatalakeAndNotify(DatalakeStatusEnum.STOPPED, "Datalake is detached", payload.getResourceId());
-                if (variables.containsKey(RESIZED_SDX)) {
-                    SdxCluster sdxCluster = (SdxCluster) variables.get(RESIZED_SDX);
-                    MDCBuilder.buildMdcContext(sdxCluster);
-                    try {
-                        transactionService.required(() -> {
-                            SdxCluster created = sdxClusterRepository.save(sdxCluster);
-                            ownerAssignmentService.assignResourceOwnerRoleIfEntitled(created.getInitiatorUserCrn(), created.getCrn(), created.getAccountId());
-                            sdxStatusService.setStatusForDatalake(DatalakeStatusEnum.REQUESTED, "Data lake requested", created);
-                            context.setSdxId(created.getId());
-                            return created;
-                        });
-                        LOGGER.info("Detaching Sdx with Name: {} and Crn: {} is complete", sdxCluster.getName(), sdxCluster.getCrn());
-                    } catch (TransactionService.TransactionExecutionException e) {
-                        throw new TransactionService.TransactionRuntimeExecutionException(e);
-                    }
+            protected void doExecute(SdxContext context, SdxDetachFailedEvent payload,
+                    Map<Object, Object> variables) throws Exception {
+                SdxCluster resized = (SdxCluster) variables.get(RESIZED_SDX);
+                SdxCluster detached = (SdxCluster) variables.get(DETACHED_SDX);
+                LOGGER.error("Failed to attach new cluster with ID {} during detaching of cluster with ID {}." +
+                        " Attempting to reattach detached cluster.", resized.getId(), detached.getId());
+
+                // Reattach detached cluster.
+                detached = sdxAttachService.reattachSdx(detached);
```

Comment:
This may be an issue with my method names, reattachSdx is the central method which calls all of the other methods in the attach service.
Let me know if you think I should rename it to make this clearer.

Reply:
I see. Please make is clear in the name.

Reply:
Changed the name lmk what you think.